bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlText.java

```diff
@@ -143,4 +143,10 @@ public int hashCode() {
     public Type getType() {
         return this.type;
     }
+
+    private boolean isXmlSequenceWithSingletonTextValue(Object o) {
+        return ((o instanceof XmlSequence)
+                && ((XmlSequence) o).isSingleton()
+                && ((XmlSequence) o).getItem(0).getNodeType() == XmlNodeType.TEXT);
```

Comment:
Break into separate checks extracting `(XmlSequence) o` out to a variable.

Reply:
Fixed with dd8e19f2a0e5255dbe863e931bffe66bf0190024